src/actions.js

```diff
@@ -47,19 +38,61 @@ export function loadMoviesFailure() {
     return { type: LOAD_MOVIES_FAILURE, error: true }
 }
 
+export function loadMoviesRequest(data) {
+    return { type: LOAD_MOVIES_REQUEST, payload: data }
+}
+
+export function loadMovieDetailsRequest(data) {
+    return { type: LOAD_MOVIEDETAILS_REQUEST, payload: data }
+}
 
 
 
-export function loadMoviesRequest(query) {
-    return function (dispatch, getState) {
-        const movieService = new MovieService();
-        const state = getState();
-        return movieService.searchMovie(state.filter.searchBy, query, state.order.orderBy, state.order.order)
-            .then(data => dispatch(loadMoviesSuccess(data)))
-            .catch(error => dispatch(loadMoviesFailure()));
+export function* moviesSaga() {
+    yield all([
+        watchFetchMovies(),
+        watchFetchMovieDetails()
+    ]);
+}
+
+export function* watchFetchMovies() {
+    yield takeLatest(LOAD_MOVIES_REQUEST, fetchMoviesAsync);
+}
+
+export function* watchFetchMovieDetails() {
+    yield takeLatest(LOAD_MOVIEDETAILS_REQUEST, fetchMovieDetailsAsync);
+}
+
+export function* fetchMoviesAsync(action) {
+
+    const movieService = yield new MovieService();
+    const filter = yield select(selectedFilter);
+    const order = yield select(selectedOrder);
+
+    try {
+        const movies = yield movieService.searchMovie(
+            filter.searchBy,
+            action.payload,
+            order.orderBy,
+            order.order)
+
+        yield put(loadMoviesSuccess(movies));
+    }
+    catch (error) {
+        yield put(loadMoviesFailure());
     }
 }
 
+export function* fetchMovieDetailsAsync(action) {
+    const movieService = yield new MovieService();
+    const movie = yield movieService.getMovie(action.payload);
```

Comment:
you need to wrap this call with
```javascript
yield call(() => movieService.getMovie(action.payload))
```

Otherwise saga won't wait for it to be resolved

Reply:
fixed please review